### PR TITLE
Remove Secure Connection macro and add the config to wifimanager.

### PR DESCRIPTION
**Pull request by h2zero**
This adds the ability to specify if the MQTT broker connection is secure via a checkbox on the config page of WiFiManager.
Also added to the WiFiManager config page is a text box to enter the brokers TLS certificate.

In the case of using manual configuration a new macro is created `MQTT_SECURE_DEFAULT`; when defined as `true` a secure connection will be requested and the certificate defined in user_config.h will be used.

### CODE REVIEW

**Comment by h2zero** [comment]
Build fails on env:nodemcuv2-2g

platformio/packages/framework-arduinoespressif8266/tools/sdk/libc/xtensa-lx106-elf/lib/libc.a(lib_a-sscanf.o): in function `sscanf':
/home/earle/src/esp-quick-toolchain/repo/newlib/newlib/libc/stdio/sscanf.c:411: multiple definition of `sscanf'; .pio/build/nodemcuv2-2g/lib621/libA6lib.a(sscanf.cpp.o):sscanf.cpp:(.text.sscanf+0x4): first defined here

Cause is the A6lib sscanf.cpp is redefining newlib functions. Seems that the file could be removed from the lib as that was included before newlib was in the 8266 framework I believe.

**Comment by 1technophile** [comment]
> Cause is the A6lib sscanf.cpp is redefining newlib functions. Seems that the file could be removed from the lib as that was included before newlib was in the 8266 framework I believe.

I would suggest to fork it, modify and point to the fork temporarily.
You may submit a PR to the A6lib repo.

File: main/User_config.h [hunk 1]
```diff
@@ -80,6 +80,10 @@ const byte ip[] = {192, 168, 1, 99};
 const byte mac[] = {0xDE, 0xED, 0xBA, 0xFE, 0x54, 0x95}; //W5100 ethernet shield mac adress
 #endif
 
+#ifndef NTP_SERVER
+#  define NTP_SERVER "pool.ntp.org"
+#endif
```
Comment (1technophile) [inline main/User_config.h:85]: Should we exclude Arduino UNO and Mega here?

Comment (h2zero) [reply main/User_config.h:85]: I didn't think it necessary since it only defines a constant. It can be moved inside a conditional check if you prefer.

**Comment by h2zero** [comment]
> 
> 
> > Cause is the A6lib sscanf.cpp is redefining newlib functions. Seems that the file could be removed from the lib as that was included before newlib was in the 8266 framework I believe.
> 
> I would suggest to fork it, modify and point to the fork temporarily.
> You may submit a PR to the A6lib repo.

done

File: main/User_config.h (hunk 1 above)
Comment (1technophile) [reply main/User_config.h:85]: ok good point

**Pull request merged**